### ISSUE TITLE
Uploading Unicoded file names can give problems.

### ISSUE THREAD

**Issue by OnikenX** [brave/brave-browser#9878]
Uploading files with unicode characters in the name on discord crashes the tab and gives the RESULT_CODE_KILLED_BAD_MESSAGE error code. 

## Steps to Reproduce

   1. create a file with unicode caracters like ç or õ in the name.
   2. click in the plus sign in discord( don't just drag, discord will not crash but it will give a error in the tab) and select the file.
   3. instant crash

## Actual result:
Shows a file with empty name and Crashes

## Expected result:
Getting the file name right like in firefox and uploads it without problem

## Reproduces how often: 
100% of the time

## Brave version (brave://version info)

Brave | 1.8.95 Chromium: 81.0.4044.138 (Official Build) unknown (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Linux
JavaScript | V8 8.1.307.32
Flash | 32.0.0.371 /usr/lib/PepperFlash/libpepflashplayer.so
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36
Command Line | /usr/lib/brave-bin/brave --ppapi-flash-path=/usr/lib/PepperFlash/libpepflashplayer.so --ppapi-flash-version=32.0.0.371 --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --sync-url=https://no-thanks.invalid --enable-features=AutoupgradeMixedContent,PassiveMixedContentWarning,MixedContentSiteSetting,SimplifyHttpsIndicator,PasswordImport,WebUIDarkMode,DnsOverHttps --disable-features=AutofillServerCommunication,TextFragmentAnchor,AllowPopupsDuringPageUnload,NotificationTriggers,LookalikeUrlNavigationSuggestionsUI,AudioServiceOutOfProcess,SmsReceiver,VideoPlaybackQuality --flag-switches-begin --load-media-router-component-extension=1 --flag-switches-end --disable-webrtc-apm-in-audio-service --disable-sync
Executable Path | /usr/lib/brave-bin/brave
Profile Path | /home/oniken/.config/BraveSoftware/Brave-Browser/Default

## Version/Channel Information:

dont know here to find the others

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
nope

- Does the issue resolve itself when disabling Brave Rewards?
nada

- Is the issue reproducible on the latest version of Chrome? 
I don't have chrome but it ~~works as expected with the discord app each is a chromium app like brave.~~ with the app it removes the especial caracters and puts in the same place ascii caracters in the chat but in the message menu with the file it reads the file just fine, wierd. but it work i think, problably it does it because of linux's utf-8 and windows's utf-16 incompatiblities.

## Miscellaneous Information:

i dont know if this is some problem with brave's linux version as again utf-8 vs 16 in windows that the devs may have broken wile playing with the chromium sources.

i tried with docs and it gives a in app error when trying to upload the same file, again, in firefox it works fine.

i dont have chrome to try these things but it should be as a brave problem if a chromium app works.

**Comment by rebron** [comment]
@OnikenX If you get a chance, can you send the crash report in brave://crashes and put the crash id in here so we can diagnose.

**Comment by OnikenX** [comment]
![Screenshot_20200629_035119](https://user-images.githubusercontent.com/25059102/85968029-cdbd9c80-b9bb-11ea-9b5c-b9ba1dec8633.png)
![Screenshot_20200629_035147](https://user-images.githubusercontent.com/25059102/85968054-e4fc8a00-b9bb-11ea-83d1-b8b75e23d2e3.png)
![Screenshot_20200629_035138](https://user-images.githubusercontent.com/25059102/85968070-eded5b80-b9bb-11ea-9c9b-2e2c6c9af36e.png)
It doesn't give me a option to send a crash report manually, is there a file where the crashes are saved?

**Comment by alexdupre** [comment]
FWIW I'm experiencing the same issue on Chromium 84 for FreeBSD. Very annoying, because the issue was quite hidden, since I was uploading the file in a AJAX call and such error wasn't so evident.

**Comment by alexdupre** [comment]
I've opened a chromium bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=1127877

**Comment by evissima** [comment]
This bug also happens with basically any file. It has been happening to me lately, with any file and any name, not necessarily with those characters, when trying to upload anywhere, Google Drive, Trello, Gmail... Gmail shows a text saying "this file is 0 bytes" or something similar. I don't know if this helps at all but just wanted to confirm it still happens.

**Comment by BertrandBordage** [comment]
@evissima I face the same issue, and I suspect it’s because Brave installed using snap has no permission to read files outside of its virtual filesystem.
This AskUbuntu answer suggests to install Brave from an APT repository instead (make sure you ran `sudo snap remove brave` before): https://askubuntu.com/a/1102561/141164

**Comment by OnikenX** [comment]
@BertrandBordage has you can see in my first post it is installed as a system bin, not containerized, so this is not a snap problem

**Comment by BertrandBordage** [comment]
@OnikenX Yes, so we are now 100% sure that there are two issues: one with the snap install with all files (@evissima’s message I was replying to) and another issue with unicode file names for some users.

**Comment by otherprod** [comment]
Hi, if it can help some of us that will fall on this issue, for me the reason of the problem was the local not setted correctly, by reset the  local to french fr_FR corrected the problem.

I had changed to fr_MA to test somethign in the past, and from there the bug appeared, i just found the solution now by folllowing the unicode you spoke about, for me the problem was same with all kinf of files not only with specific caracters.

Solved by setting :

![image](https://user-images.githubusercontent.com/3120057/123513132-457fdf00-d683-11eb-905d-80548f661dfe.png)

**Comment by raulcesar** [comment]
I just ran into this error with brave 1.38.119, but have also confirmed the problem in Chromium Version 102.0.5005.61 (Official Build) Arch Linux (64-bit).
To replicate, just need to create a very simple HTML, like so:
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <h1>Upload File</h1>
    <input type="file" />
  </body>
</html>
```
Open the file directly in the browser or serve the file with something like: 
```sh
python3 -m http.server 9000
```
Next, create a file like so:
```sh
touch testã.txt
```
Try to upload, and: 

![image](https://user-images.githubusercontent.com/1403840/171900603-545ed99a-095c-42cb-a2db-63c5db48afc6.png)

To see that the problem is in the name, do:
```sh
mv testã.txt testa.txt
```
and try the upload again, and:
![image](https://user-images.githubusercontent.com/1403840/171900927-fef6c409-2fe6-49d0-9931-bf291e7106f5.png)

**Comment by raulcesar** [comment]
Just a follow-up to previous post, I noticed that on my home computer, the tests for uploading files did not occur. 

Upon further investigation, I noticed that the local settings (i.e. variables like LANG, LC_ALL, etc.) were a bit different, so I started playing around with them. 

I noticed that if I explicitly set than **LANG** to something with ".UTF-8" (so en_US.UTF-8 or pt_BR-UTF-8), the error would go away. If I unset the variable, or set it to something like "LANG=C", the error returned.... however, event with then LANG setting to "C" (or unset), I was able to make the error go away by setting **LC_ALL** (or even **LC_CTYPE**) to something with ".UTF-8" (i.e. en_US.UTF-8, pt_BR-UTF-8, etc.)

So, in my case, to get everything working, I just changed my locale settings (in /etc/locale.conf) to:
```sh
LANG=C
LANGUAGE=en
LC_ALL=en_US.UTF-8
```

Hope this helps.

**Comment by mjholub** [comment]
Still facing this in 2024 with Artix GNU/Linux and non-ASCII filenames

**Comment by georgethemarmot** [comment]
The issue seems to be also happening when path is containing Unicode characters
https://github.com/brave/brave-browser/issues/51753
Will eventually check on [raulcesar](https://github.com/raulcesar) locales solutions
